client/state/initial-state.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+import localforage from 'localforage';
+
+/**
+ * Internal dependencies
+ */
+import { createReduxStore } from 'state';
+import { reducer } from 'state';
+import { TO_OBJECT, FROM_OBJECT } from 'state/action-types'
+
+/**
+ * Module variables
+ */
+const debug = debugModule( 'calypso:state' );
+
+const localforageConfig = {
+	name: 'calypso',
+	storeName: 'calypso_store',
+	description: 'Calypso Storage'
+};
+
+function toObject( state ) {
+	return reducer( state, { type: TO_OBJECT } );
+}
+
+function fromObject( state ) {
+	return reducer( state, { type: FROM_OBJECT } );
+}
+
+function loadInitialState( initialState ) {
+	debug( 'loading initial state with', initialState );
+	return createReduxStore( fromObject( initialState ) );
```

Comment:
Did you find that there was any issue using `dispatch` with these initialization actions? Would be nice to not have to imitate it in `toObject`/`fromObject` if we can avoid it, instead:

```js
const store = createReduxStore();
store.dispatch( fromObject( initialState ) );
return store;
```